resdk/analysis/differential_expressions.py

```diff
@@ -0,0 +1,107 @@
+"""Differential expressions analysis."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from resdk.resources.utils import (
+    get_data_id, get_resolwe, get_resource_collection, get_samples, is_collection, is_relation,
+)
+
+__all__ = ('cuffdiff', )
+
+
+def cuffdiff(resource, annotation, genome=None, multi_read_correct=None, fdr=None,
+             library_type=None, library_normalization=None, dispersion_method=None, threads=None):
+    """Run Cuffdiff_ for selected cuffquants.
+
+    This method runs `Cuffdiff`_ process with ``annotation`` specified
+    in arguments. Library type is by defalt fr-unstranded. Other parameters
+    defaults: multi_read_correct=false, fdr=0.05, library_normalization=geometric,
+    dispersion_method=pooled, threads=1. Parameter genome is optional.
+
+    .. _Cuffdiff:
+        http://resolwe-bio.readthedocs.io/en/latest/catalog-definitions.html#process-cuffdiff
+
+    :param annotation: annotation file
+    :type annotation: `~resdk.resources.data.Data`
+    :param genome: genome object to use for bias detection and
+        correction algorithm
+    :type genome: `~resdk.resources.data.Data`
+    :param bool multi_read_correct: do initial estimation procedure to
+        more accurately weight reads with multiple genome mappings
+    :param fdr: the allowed false discovery rate
+    :type fdr: decimal
+    :param str library_type: options are fr-unstranded, fr-firststrand,
+        fr-secondstrand
+    :param str library_normalization: options are geometric, classic-fpkm,
+        quartile
+    :param str dispersion_method: options are pooled, per-condition,
+        blind, poisson
+    :param int threads: use this many processor threads
+
+    """
+    inputs = {'annotation': get_data_id(annotation)}
+    if genome is not None:
+        inputs['genome'] = genome
+    if multi_read_correct is not None:
+        inputs['multi_read_correct'] = multi_read_correct
+    if fdr is not None:
+        inputs['fdr'] = fdr
+    if library_type is not None:
+        inputs['library_type'] = library_type
+    if library_normalization is not None:
+        inputs['library_normalization'] = library_normalization
+    if dispersion_method is not None:
+        inputs['dispersion_method'] = dispersion_method
+    if threads is not None:
+        inputs['threads'] = threads
+
+    relation_filter = {}
+    collection_id = get_resource_collection(resource)
+    if collection_id:
+        relation_filter['collection'] = collection_id
+
+    samples = get_samples(resource)
+
+    input_objects = [annotation]
+    input_objects.extend(samples)
+    resolwe = get_resolwe(*input_objects)
+
+    relations = resolwe.relation.filter(
```

Comment:
This query will pull all case-control relations that are in any way related to the samples (i.e. any sample from the resource is in this relation). This is further connected to skipping unsuitable relations mentioned later. But most important: all `case-control` relations from the server will NOT be pulled.